### ISSUE TITLE
bug: Step Functions describe-activity incorrectly rejects ARN

### ISSUE THREAD

**Issue by peter-smith-phd** [localstack/localstack#11916]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When running `describe-activity` on an activity ARN that contains punctuation symbols in the name, an `Invalid arn` error is reported.

```
$ awslocal stepfunctions create-activity --name RetryParallelStateIntegrationTest.2024-11-23-06-19-53.60e0ecac-0dbb-4e5e-bddb-c
{
    "activityArn": "arn:aws:states:ca-west-1:000000000000:activity:RetryParallelStateIntegrationTest.2024-11-23-06-19-53.60e0ecac-0dbb-4e5e-bddb-c",
    "creationDate": 1732374466.340198
}

$ awslocal stepfunctions describe-activity --activity-arn arn:aws:states:ca-west-1:000000000000:activity:RetryParallelStateIntegrationTest.2024-11-23-06-19-53.60e0ecac-0dbb-4e5e-bddb-c

An error occurred (InvalidArn) when calling the DescribeActivity operation: Invalid arn: 'arn:aws:states:ca-west-1:000000000000:activity:RetryParallelStateIntegrationTest.2024-11-23-06-19-53.60e0ecac-0dbb-4e5e-bddb-c'
```

### Expected Behavior

The correct response (from AWS) is as follows:

```
$ aws stepfunctions create-activity --name RetryParallelStateIntegrationTest.2024-11-23-06-19-53.60e0ecac-0dbb-4e5e-bddb-c
{
    "activityArn": "arn:aws:states:ca-west-1:491085397214:activity:RetryParallelStateIntegrationTest.2024-11-23-06-19-53.60e0ecac-0dbb-4e5e-bddb-c",
    "creationDate": 1732373726.378
}

$ aws stepfunctions describe-activity --activity-arn arn:aws:states:ca-west-1:491085397214:activity:RetryParallelStateIntegrationTest.2024-11-23-06-19-53.60e0ecac-0dbb-4e5e-bddb-c
{
    "activityArn": "arn:aws:states:ca-west-1:491085397214:activity:RetryParallelStateIntegrationTest.2024-11-23-06-19-53.60e0ecac-0dbb-4e5e-bddb-c",
    "name": "RetryParallelStateIntegrationTest.2024-11-23-06-19-53.60e0ecac-0dbb-4e5e-bddb-c",
    "creationDate": 1732373726.378,
    "encryptionConfiguration": {
        "type": "AWS_OWNED_KEY"
    }
}
```

### How are you starting LocalStack?

With the `localstack` script

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

    localstack start

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

    see above

### Environment

```markdown
- OS: MacOS Sequoia
- LocalStack:
  LocalStack version: 4.0.2.dev3
  LocalStack build date: 2024-11-23
  LocalStack build git hash: 39ac0b968
```

**Comment by MEPalma** [comment]
Thank you for taking the time to compile this report! I was able to replicate the behaviour that resulted in the failure. We recently merged some changes that aim to address this issue. These changes are available since v4.0.3. I would be grateful if you could test the new build at your earliest convenience and provide feedback on whether it resolves the problem you encountered. Thank you once again for bringing this issue forward.

**Comment by peter-smith-phd** [comment]
Thanks Marco, it now works for me :-)
